ecosystem/gstreamer_plugin/README.md

```diff
@@ -0,0 +1,408 @@
+
+```bash
+export OUTPUT="path_to_the_file_we_want_to_save"
+```
+
+#### 3.2.3. Pipline example for multicast with `y210` format
+
+To run the `mtl_st20p_rx` plugin, use the following command to specify the input parameters of the incoming stream.
+
+> **Warning**: To receive data, ensure that a transmission is running with the same parameters on the `239.168.75.30` address.
+
+```bash
+# If you don't know how to find the VFIO PCI address of your device
+# refer to Media-Transport-Library/doc/run.md
+export VFIO_PORT_R="pci_address_of_the_device"
+
+# Run the receiver pipeline y210 FHD 60fps on port 20000
+gst-launch-1.0 mtl_st20p_rx rx-queues=4 udp-port=20000 payload-type=112 dev-ip="192.168.96.3" ip="239.168.75.30" dev-port=$VFIO_PORT_R rx-pixel-format=v210 rx-height=1080 rx-width=1920 rx-fps=25 ! \
+filesink location=$OUTPUT --gst-plugin-path $GSTREAMER_PLUGINS_PATH
+```
+
+This command sets up the receiver pipeline with the specified parameters and saves the received video to the specified output path.
+
+## 4. SMPTE ST 2110-30 Raw audio plugins
+
+Audio plugins for MTL that are able to send, receive synchronous raw audio via the MTL pipeline API.
+
+### 4.1. Running the SMPTE ST 2110-30 transmission plugin mtl_st30p_tx
+
+#### 4.1.1. Supported Parameters and Pad Capabilities
+
+The `mtl_st30p_tx` plugin supports the following pad capabilities:
+
+- **Formats**: `S16LE`, `S24LE`, `S32LE`
+- **Sample Rate Range**: 44100, 48000, 96000
+- **Channels Range**: 1 to 8
+
+**Arguments**
+| Property Name       | Type   | Description                                           | Range                   | Default Value |
+|---------------------|--------|-------------------------------------------------------|-------------------------|---------------|
+| retry               | uint   | Number of times the MTL will try to get a frame.      | 0 to G_MAXUINT          | 10            |
+| tx-samplerate       | uint   | Sample rate of the audio.                             | [gst_mtl_supported_audio_sampling](#232-supported-audio-sampling-rates-gst_mtl_supported_audio_sampling) | 0 |
+| tx-channels         | uint   | Number of audio channels.                             | 1 to 8                  | 2             |
+
+#### 4.1.2. Example GStreamer Pipeline for Transmission with s16LE format
+
+To run the `mtl_st30p_tx` plugin, you need to setup metadata (Here we are using pipeline capabilities).
+Instead of using input video we opted for build-in GStreamer audio files generator.
+
+```bash
+# If you don't know how to find the VFIO PCI address of your device
+# refer to Media-Transport-Library/doc/run.md
+export VFIO_PORT_T="pci_address_of_the_device"
+
+# Audio pipeline with 48kHz sample rate on port 30000
+gst-launch-1.0 audiotestsrc ! \
+audio/x-raw,format=S16LE,rate=48000,channels=2 ! \
+mtl_st30p_tx tx-queues=4 rx-queues=0 udp-port=30000 payload-type=113 dev-ip="192.168.96.3" ip="239.168.75.30" dev-port=$VFIO_PORT_T \
```

Comment:
do we really need to specify rx-queues in tx mode?

Reply:
seems like some imperfection of the design. Maybe we could pass something to gst_mtl_common_init_handle to specify mode (TX or RX)?

Reply:
That could as well be 

I assumed 16 to be a sweet spot we can change it later mby during the common handle patch :<

Reply:
Basically, when we will have the common handle we can decide what to do with the amount of queues but for now i would leave the 16 as a safe number ^^

Reply:
![image](https://github.com/user-attachments/assets/5e7c5a9f-9b4b-45c3-8ac1-2aa6d1b93f53)